### PR TITLE
MINIFICPP-2309 Add JSON flow config examples

### PR DESCRIPTION
**Pull request by lordgamez**
- Also add a few new use case examples

https://issues.apache.org/jira/browse/MINIFICPP-2309

----------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: examples/tailfile_config.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
+    "rootGroup": {
+        "name": "MiNiFi Flow",
+        "processors": [
+            {
+                "name": "Log tailfile attributes",
+                "identifier": "41141af1-3c43-4fa4-8236-2b207b0fe9d8",
+                "type": "org.apache.nifi.minifi.processors.LogAttribute",
+                "schedulingStrategy": "TIMER_DRIVEN",
+                "schedulingPeriod": "1000 ms",
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply examples/tailfile_config.json]: I think it should be event driven as we recommend using that in this case, there are other occurrences of timer driven cases. Updated in d67362d317cc2369536a9eede32709da58541444

File: examples/README.md [hunk 2]
```diff
@@ -37,122 +43,158 @@ The following examples show simple flow configurations for several common use ca
...
 
-The flow: ConsumeJournald &#10132; LogAttribute
+The flow: [ConsumeJournald](../PROCESSORS.md#consumejournald) &#10132; [LogAttribute](../PROCESSORS.md#logattribute)
 
 ## HTTP Operations
 
 ### HTTP POST Invocation
 
-Using the [http_post_config.yml](http_post_config.yml) flow configuration MiNiFi transfers flowfile data received from the GetFile processor by invoking an HTTP endpoint with POST method.
+Using the [http_poste_config.yml](http_post_config.yml)/[http_post_config.json](http_post_config.json)/[http_post_config.nifi.schema.json](http_post_config.nifi.schema.json) flow configuration MiNiFi transfers flowfile data received from the GetFile processor by invoking an HTTP endpoint with POST method.
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply examples/README.md]: Fixed in d67362d317cc2369536a9eede32709da58541444

**Review by fgerlits — APPROVED** [review]

File: examples/splittext_puttcp_config.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    "Input Directory": "/tmp/input"
+                },
+                "autoTerminatedRelationships": []
+            },
+            {
+                "name": "Split text by lines skipping the first 3 header lines",
+                "identifier": "40b45f52-ae1d-443c-9465-7e070ce3bea0",
+                "type": "org.apache.nifi.processors.standard.SplitText",
+                "schedulingStrategy": "EVENT_DRIVEN",
+                "schedulingPeriod": "1 sec",
```
Comment (szaszm) [inline examples/splittext_puttcp_config.json:23]: I think we should remove the scheduling period setting from EVENT_DRIVEN processors.

Comment (lordgamez) [reply examples/splittext_puttcp_config.json]: Updated in dcab8ffd0883e3fc881e85f6a30fcbdf0cb7e9f2

File: examples/getfile_putfile_config.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "name": "MiNiFi Flow",
+        "processors": [
+            {
+                "name": "GetFile",
+                "identifier": "b01793b6-5275-4ee3-9fc0-d9e39ce3162f",
+                "type": "org.apache.nifi.processors.standard.GetFile",
+                "schedulingStrategy": "TIMER_DRIVEN",
+                "schedulingPeriod": "2 sec",
+                "properties": {
+                    "Input Directory": "/tmp/input"
```
Comment (adamdebreceni) [inline examples/getfile_putfile_config.json:12]: this seems to be different from its yaml counterpart, multiple properties seem to be missing for both processors

Comment (lordgamez) [reply examples/getfile_putfile_config.json]: Good catch, fixed in 20feddbeb43868d968be1a932939a2539c653eb8

File: examples/querydbtable_config.nifi.schema.json [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+                    "y": 0.0
+                },
+                "type": "org.apache.nifi.minifi.sql.controllers.ODBCService",
+                "bundle": {
+                    "group": "org.apache.nifi.minifi",
+                    "artifact": "minifi-sql",
+                    "version": "0.15.0"
+                },
+                "properties": {
+                    "Connection String": "enc{7YHB8DDp3xI5wiKGfsoflVELCXpY7M/v||qKzN8vELnMZGysNDRWitok+XHy0pXkwKF9oR0nNoUet/AfE9aj0oOBHJppSCVKPlq0WjbM82+4HYY3hLcfPssHFn2/zeR1vtDo3wRHptoFmG9pncr8O8noH2tPXHKgqWJnYtwNAiGhEQLuqTSnNXjBe5KSgKNatq}"
```
Comment (szaszm) [inline examples/querydbtable_config.nifi.schema.json:219]:
```suggestion
                    "Connection String": "Driver={PostgreSQL ANSI};Server=postgresql-server;Port=5432;Database=postgres;Uid=postgres;Pwd=password;"
```

**Review by szaszm — APPROVED** [review]
I'd keep the sensitive value unencrypted in the example, but it's fine either way.

File: examples/querydbtable_config.nifi.schema.json (hunk 5 above)
Comment (lordgamez) [reply examples/querydbtable_config.nifi.schema.json]: Good point, updated in 3e592636133c986a3557dbec8929929ba3617b04

**Pull request closed without merging**